### PR TITLE
Audit API Extractor error strings

### PR DESCRIPTION
**Pull request by pgonzal**
I reviewed all the `reportWarning()` and `reportError()` calls, and cleaned up the error messages.

### CODE REVIEW

File: libraries/api-extractor/src/ApiDefinitionReference.ts [hunk 1]
```diff
@@ -131,11 +131,12 @@ export default class ApiDefinitionReference {
...
     } else {
       // the export name is required
-       throw reportError(`Api reference expression contains invalid characters: ${apiReferenceExpr}`);
+       throw reportError(`The API item reference contains an invalid "exportName.memberName"`
+        + ` expression: "${apiReferenceExpr}"`);
     }
 
     if (!apiReferenceExpr.match(ApiDefinitionReference._exportRegEx)) {
-      throw reportError(`Api reference expression contains invalid characters: ${apiReferenceExpr}`);
+      throw reportError(`API item reference contains invalid characters: "${apiReferenceExpr}"`);
```
Comment (chackman) [inline libraries/api-extractor/src/ApiDefinitionReference.ts]: >API item reference [](start = 25, length = 18)

Please start with "The API item reference"

File: libraries/api-extractor/src/definitions/ApiDocumentation.ts [hunk 2]
```diff
@@ -333,37 +334,37 @@ export default class ApiDocumentation {
...
         tokenizer.getToken();
         // This would be a program bug
-        this.reportError(`Unexpected token: ${token.type} ${token.tag} ${token.text}`);
+        this.reportError(`Unexpected token: ${token.type} ${token.tag} "${token.text}"`);
       }
     }
 
     if (releaseTagCount > 1) {
-      this.reportError('More than one release tag was specified');
+      this.reportError('More than one one release tag (@alpha, @beta, etc) was specified');
```
Comment (chackman) [inline libraries/api-extractor/src/definitions/ApiDocumentation.ts]: >one one [](start = 34, length = 7)

There's more than one "one" in this message ;-)

File: libraries/api-extractor/src/DocElementParser.ts [hunk 3]
```diff
@@ -195,7 +197,8 @@ export default class DocElementParser {
     if (linkDocElement && pipeSplitContent.length > 1) {
       const displayTextParts: string[] = pipeSplitContent[1].match(this._wordRegEx);
       if (displayTextParts && displayTextParts[0].length !== pipeSplitContent[1].length) {
```
Comment (pgonzal) [inline libraries/api-extractor/src/DocElementParser.ts]: @dagaeta Not sure what this code was actually trying to validate

File: libraries/api-extractor/src/DocElementParser.ts [hunk 4]
```diff
@@ -159,7 +160,8 @@ export default class DocElementParser {
 
       // Make sure only a single url is given
       if (urlContent.length > 1 && urlContent[1] !== '' ) {
-        documentation.reportError('Invalid @link parameter, url must be a single string.');
+        documentation.reportError('The {@link} tag contains additional spaces after the URL;'
```
Comment (pgonzal) [inline libraries/api-extractor/src/DocElementParser.ts]: I made it URL everywhere

---
In reply to: [122554191](https://github.com/Microsoft/web-build-tools/pull/239#discussion_r122554191) [](ancestors = 122554191)

Comment (chackman) [inline libraries/api-extractor/src/DocElementParser.ts]: Great, thanks!

---
In reply to: [122771613](https://github.com/Microsoft/web-build-tools/pull/239#discussion_r122771613) [](ancestors = 122771613,122554191)

**Comment by chackman** [comment]
:shipit:

File: libraries/api-extractor/src/DocElementParser.ts (hunk 3 above)
Comment (pgonzal) [inline libraries/api-extractor/src/DocElementParser.ts]: We discussed it.  I will address it in a separate PR

---
In reply to: [122555107](https://github.com/Microsoft/web-build-tools/pull/239#discussion_r122555107) [](ancestors = 122555107)

**Comment by pgonzal** [comment]
:shipit:

**Pull request merged**